### DISCUSSION TITLE
Unkown character format in extracted content

### DISCUSSION THREAD

**Discussion by drrobotic** [pdfcpu/pdfcpu#725] [General]
Hi,
i wanted to extract raw content of some PDFs i got and for older ones its working just fine,
but i have newer ones where the text is not normal readable ASCII, instead another format 
which i couldnt really find out what it is.

Older ones content looks like:

`BT
/F0 10.5 Tf
1 0 0 1 375.591 558.63 Tm
(31.)Tj
1 0 0 1 393.08499 558.63 Tm
(Mai)Tj
1 0 0 1 412.91 558.63 Tm
(2023)Tj
`
Newer ones look like this (PDFs are similar and its the same position)

`BT
/F0 10.5 Tf
1 0 0 1 375.591 558.63 Tm
(\000\026\000\024\000\021)Tj
1 0 0 1 393.08499 558.63 Tm
(\000$\000X\000J\000X\000V\000W)Tj
1 0 0 1 428.66 558.63 Tm
(\000\025\000\023\000\025\000\026)Tj
`

Does anybody know how i decode this stuff?

**Comment by hhrutter** [comment]
Hello!
Are you using the extract content command?
Can you share a small sample for investigation?

**Comment by drrobotic** [comment]
i use the pdfcpu.ExtractPageContent(...) method the get the content in my program. if you look at the snippets above, there should be the texts "31." "August" "2023", but i get "\000\026\000\024\000\021" "\000$\000X\000J\000X\000V\000W" "\000\025\000\023\000\025\000\026"

**Comment by hhrutter** [comment]
The command returns the raw page content in PDF syntax.
In your case you have octal values representing glyph codes.
The decoding involves knowing what kind of font is used and which encoding.

**Reply by drrobotic** [reply]
ok thanks, can i find an example of decoding in your project?

**Comment by hhrutter** [comment]
Sorry not at the moment,
What you are looking for is text extraction, see #122

**Comment by hhrutter** [comment]
Closing this as answered.

ExtractPageContent returns raw PDF page content syntax, not decoded Unicode text. In this case the string operands contain encoded glyph codes, and decoding them correctly requires interpreting the font, encoding, and possibly ToUnicode CMaps.

The broader feature request for actual text extraction is tracked in #122.
